cms/djangoapps/contentstore/views/transcripts_ajax.py

```diff
@@ -208,7 +208,6 @@ def upload_transcripts(request):
         if not edx_video_id:
             edx_video_id = create_external_video(display_name='external video')
             video.edx_video_id = edx_video_id
-            video.save_with_metadata(request.user)
```

Comment:
@navinkarkera If I understand correctly, you're removing this to:
1. Reduce duplication
2. Reduce the number of calls to DB

However, I think there is a downside to this change - imagine that `create_external_video` is called, but then the `try` block below throws an error.

What do you think about the following suggestions:

Keep this line, as the downsides are not that bad:
1. Duplication in this case is compensated by the fact that all the lines related to "create external video if doesn't already exist" are going to be kept together, which reduces the complexity, and avoids introducing the bugs in the future
2. Performance drawbacks are negligible, because I would imagine that this branch would not be executed that often (though it's based on my gut feeling, not stats), and only once per video

Alternatively, you could still remove this line here, and move `video.save_with_metadata` on line 234 to the `finally` block in the `try-except` block below.

Reply:
>However, I think there is a downside to this change - imagine that create_external_video is called, but then the try block below throws an error.

@Cup0fCoffee I did think about this but still went ahead as I did want anything to be saved till the operation was successful but now looking at the code in [edxval.create_external_video](https://github.com/openedx/edx-val/blob/bcd83ee0c87fa2a80578ddaa6cad884c7ac5961d/edxval/api.py#L78-L131) I think we need to save here as it is actually creating something in the database and fetching the id.